FoodApp/Startup.cs

```diff
@@ -72,6 +73,88 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                     pattern: "{controller=Home}/{action=Index}/{id?}");
                 endpoints.MapRazorPages();
             });
+
+
+            //run migrations automatically?
+            var context = serviceProvider.GetRequiredService<ApplicationDbContext>();
+            context.Database.Migrate();
+            CreateRoles(serviceProvider);
         }
+        private void CreateRoles(IServiceProvider serviceProvider)
+        {
+
+            var roleManager = serviceProvider.GetRequiredService<RoleManager<IdentityRole>>();
+            var userManager = serviceProvider.GetRequiredService<UserManager<IdentityUser>>();
+            Task<IdentityResult> roleResult;
+            //add the roles if they don't exist
+            string[] roleNames = { "Administrator", "Restaurant", "User" };
+            foreach (var roleName in roleNames)
+            {
+                Task<bool> roleExists = roleManager.RoleExistsAsync(roleName);
+                roleExists.Wait();
+
+                if (!roleExists.Result)
+                {
+                    roleResult = roleManager.CreateAsync(new IdentityRole(roleName));
+                    roleResult.Wait();
+                }
+            }
+
+            
+            CreateUser(userManager, "admin@admin", "Administrator");
+            CreateUser(userManager, "user@user", "User");
+            CreateRestaurant(userManager, "restaurant@restaurant", "Restaurant", "Pending");
+            CreateRestaurant(userManager, "restaurant2@restaurant2", "Restaurant", "Confirmed");
+            CreateRestaurant(userManager, "restaurant3@restaurant3", "Restaurant", "Pending");
+
+        }
+        private void CreateUser(UserManager<IdentityUser> userManager, string email, string role)
+        {
+            Task<IdentityUser> testUser = userManager.FindByEmailAsync(email);
+            testUser.Wait();
+
+            if (testUser.Result == null)
+            {
+                
+                IdentityUser user = new IdentityUser();
+                user.Email = email;
+                user.UserName = email;
```

Comment:
Is this here correct? Seems awfully bold to just have the username the same as the email. May as well just use Email for identification.

Reply:
Not relevant. Mock data that isn't used wouldn't be used in production